File: src/db-gwent-server-scala/src/main/scala/com/entities/PointGroup.scala
package com.entities

import scala.collection.mutable.ListBuffer

case class PointGroup(
  originalValue: Int = 0,
  modifiers: ListBuffer[PointGroupModifier] = ListBuffer()):

  def getCalculatedValue: Int =
    modifiers.foldLeft(originalValue):
      (newPoints, modifier) => modifier.modify(newPoints)

File: src/db-gwent-server-kotlin/src/main/kotlin/com/entities/PointGroup.kt
package com.entities

class PointGroup(
    val originalValue: Int = 0,
    val modifiers: MutableList<PointGroupModifier> = mutableListOf()) {

    fun getCalculatedValue(): Int {
        return modifiers.fold(originalValue) {
            newPoints, modifier -> modifier.modify(newPoints)
        }
    }
}